swift/ql/lib/codeql/swift/security/SensitiveExprs.qll

```diff
@@ -80,13 +85,34 @@ private string regexpProbablySafe() {
   result = "(?is).*(file|path|url|invalid).*"
 }
 
+/**
+ * Gets a string that is to be tested for sensitivity.
+ */
+cached
+private string sensitiveCandidateStrings() {
+  result = any(VarDecl v).getName()
+  or
+  result = any(Function f).getShortName()
+  or
+  result = any(Argument a).getLabel()
+}
+
+/**
+ * Gets a string from the candidates that is sensitive.
+ */
+cached
```

Comment:
What is this `cached` meant to achieve? And similarly for the `cached` annotation on `sensitiveCandidateStrings`. The annotations `cached` and `private` rarely go together since `cached` mostly means "don't specialize the predicate based on call-sites". I think replacing `cached` with `pragma[nomagic]` would make this a lot less controversial.

Reply:
You're right, this `cached` annotation is not very principled, it was put there just to be sure nothing unexpected happened.  I could downgrade it to `pragma[nomagic]`, but in fact removing it entirely - along with the one a few lines down - doesn't seem to affect performance (locally).  So I've done that.  I will re-run DCA to confirm.